Automattic/wp-calypso
# Signup: Add Site Title step

This PR adds a Site Title step to the signup flows. It will allow the user to set their site title early in signup and it will give us the option to automatically fire domain suggestions based on what the user entered as a site title.

## Design mockup:

![artboard](https://cloud.githubusercontent.com/assets/184938/17690286/3a086562-6396-11e6-8b9f-9e2e23da5df0.png)

## Testing instructions:

Prerequisites: Apply D2525-code 

1. Checkout branch or use the `calypso.live` branch.
2. Go to `/start/sitetitle`.
3. Choose something in the Survey and you will land on the Site Title step.
4. Make sure the step works without breaking anything. 
5. Complete signup
6. Check if the site title on the newly created site is the one that you typed during signup.

Test live: https://calypso.live/start/sitetitle/?branch=add/site-title-step-in-signup

cc @michaeldcain @coreh @meremagee
